### ISSUE TITLE
SelectBooleanCheckbox: in Firefox, NVDA just reads "blank" when checkbox is focused with tab

### ISSUE THREAD

**Issue by Weavelor** [primefaces/primefaces#11577]
### Describe the bug

When you use tab to select a checkbox, NVDA simply reads "blank." You then need to press the arrow down key to hear the itemLabel of the checkbox. In both cases, you can press space to check or uncheck the checkbox, so it's unclear where the focus lies the first time the screen reader reads "blank." When you click the checkbox with the mouse, it always says "blank," regardless of the checkbox's state.

This issue seems to only arise when using Firefox with NVDA:

- NVDA and selectBooleanCheckbox function correctly with other browsers like Chrome or Edge.
- Other screen readers, such as Microsoft Narrator, function properly with selectBooleanCheckbox in Firefox.
- Other checkboxes, like those from [Ant Design](https://ant.design/components/checkbox), work with NVDA and Firefox.

This problem might stem from Firefox or NVDA rather than PrimeFaces, so I'd like to apologize in advance if this is not the appropriate place to address it. However, since I'm unsure where the issue lies, any help would be greatly appreciated.

### Reproducer

Can be reproduced in the [Primefaces Showcase](https://www.primefaces.org/showcase/ui/input/booleanCheckbox.xhtml?jfwid=918e0) using Firefox (123.0.1) and NVDA (2023.3.4).

### Expected behavior

NVDA should behave the same in Firefox as in Chrome or Edge

### PrimeFaces edition

Community

### PrimeFaces version

whatever the showcase uses

### JSF implementation

All

### JSF version

-

### Java version

-

### Browser(s)

Firefox 123.0.1 (64-Bit)

**Comment by melloware** [comment]
If it is just one screenreader for one browser and ALL others work I am hesitant to investigate or do anything about it from PrimeFaces point of view. There are at least 4 major screenreaders such as JAWS, NVDA, etc and 3 major browsers Chrome, Firefox, Safari.

**Comment by Bukama** [comment]
> cc @Bukama thoughts?

There's something on my mind that we had an similar issue noticed (not sure if exactly). Will check tomorrow when back at work.

**Comment by Bukama** [comment]
We had something similar: We had checkboxes in a table (to select multiple lines) and then the checkboxes' aria-label was not read out, but as far as I understand the bug description of our internal bug report that was caused by the aria-label was not rendered correctly due the `DataTableRenderer`around it. My team mate solved this by overwriting the `DataTableRenderer`
to pass-thorugh the aria-label via `pt:aria-label`. 

To come back to this: As the aria label is clearly there I see this more as an issue in the combination of Firefox and NVDA. @Weavelor Could you add a screenshot from the text output mode of the NVDA? I mean we assume that there the value is missing, but we then know that the HTML is correct but the NVDA output is not and can cleary say that this is an NVDA issue to report in their github (where they already have more than 2000 open checkbox issues https://github.com/nvaccess/nvda/issues?q=is%3Aopen+is%3Aissue+checkbox

**Comment by melloware** [comment]
Agreed I feel like this is a report you will have to make to NVDA

**Comment by Weavelor** [comment]
@Bukama certainly I can provide you with the exact outputs:

Chrome after focus with tab:
"Basic  check box  not checked "
arrow down key:
"clickable"
arrow down:
"clickable  Basic"
arrow down:
"heading    level 5  Remote"

Firefox after focus with tab:
"blank"
arrow down:
"clickable  Basic"
arrow down:
"heading    level 5  Remote"

I inspected the checkBox in both Chrome and Firefox and they look exactly like @melloware posted earlier, so maybe something weird happens when NVDA interprets the xhtml (or does the browser tell NVDA what to read? I'm not sure how this works internally). 

I will submit an issue to NVDA. Thank you all for your help and investigation.

**Comment by melloware** [comment]
@Weavelor when you submit the NVDA issue you can refer back to this ticket so its linked.  I will close this for now.

**Comment by melloware** [comment]
@Weavelor @f1nnk  i am re-opening this.  This CSS style PrimeFaces has had since 2011.

```css
.ui-helper-hidden-accessible {
    border: 0;
    clip: rect(0 0 0 0);
    height: 1px;
    margin: -1px;
    overflow: hidden;
    padding: 0;
    position: absolute;
    width: 1px;
}
```

I found this stack overflow of a more modern version of it that is better supposedly see:
https://stackoverflow.com/a/62109988/502366

**Comment by melloware** [comment]
Tracking in NVDA ticket.
